### ISSUE TITLE
Segfault on consuming messages via Queue

### ISSUE THREAD

**Issue by kwn** [php-rdkafka/php-rdkafka#53]
Hi, 

I've just spotted some issue around consuming messages using Queue object. My environmental settings:

```
php version: 5.6.20
librdkafka: 0.8.6.0
php-rdkafka: the recent one (commit 5ff585986c0cbf6f01753f79b5c2291d437a60eb)
```

My Kafka broker runs with default configuration. There are three topics, every topic has stored ~50 messages, so there's no uncommon settings or edge cases here. Once I'm running a simple unit test I get a segfault. A code for unit test:

``` php
<?php

namespace RdKafka;

class QueueTest extends \PHPUnit_Framework_TestCase
{
    const PARTITION = 0;

    /**
     * @var ConsumerTopic
     */
    private $consumerTopic;

    /**
     * @var Queue
     */
    private $queue;

    public function setUp()
    {
        $consumer = new Consumer();
        $consumer->addBrokers('localhost:9092');

        $this->consumerTopic = $consumer->newTopic('test');
        $this->queue = $consumer->newQueue();
    }

    public function testConsumeIntoQueue()
    {
        $this->consumerTopic->consumeQueueStart(self::PARTITION, -5, $this->queue);
        $this->consumerTopic->consume(self::PARTITION, 100);
        $this->consumerTopic->consumeStop(self::PARTITION);

        $message = $this->queue->consume(200);
    }
}

```

Once I run tests, I get a below output:

```
*** rdkafka_msg.c:40:rd_kafka_msg_destroy: assert: rk->rk_producer.msg_cnt > 0 ***
rd_kafka_t 0x2e77bd0: rdkafka#consumer-28
 refcnt 5, producer.msg_cnt 0
 rk_rep reply queue: 0 ops
 brokers:
 rd_kafka_broker_t 0x2ebae10: localhost:9092/bootstrap NodeId -1 in state UP (for 0.013s)
  refcnt 1
  outbuf_cnt: 0 waitresp_cnt: 0
  1 messages sent, 25 bytes, 0 errors, 0 timeouts
  1 messages received, 150 bytes, 0 errors
  0 messageset transmissions were retried
  0 toppars:
 rd_kafka_broker_t 0x7ff16c001fc0: thinkpad:9092/0 NodeId 0 in state UP (for 0.012s)
  refcnt 2
  outbuf_cnt: 0 waitresp_cnt: 0
  1 messages sent, 25 bytes, 0 errors, 0 timeouts
  1 messages received, 150 bytes, 0 errors
  0 messageset transmissions were retried
  1 toppars:
Segmentation fault (core dumped)
```

**Comment by arnaud-lb** [comment]
Hi @kwn!

Thanks for reporting this.

I couldn't reproduce the issue. I've tried running the script and got this error:

```
RdKafka\Exception: Local: Invalid argument or configuration
```

This particular error is caused by the negative offset passed to `consumeQueueStart()`. I've tried with `RD_KAFKA_OFFSET_BEGINNING` and `rd_kafka_offset_tail(5)`; this eliminates this error, but fails to reproduce the original issue.

Is this something that is happening every time on your side ?

The script is consuming from the topic directly: `$this->consumerTopic->consume(self::PARTITION, 100);` after having told the topic to move consumed messages to a queue: `$this->consumerTopic->consumeQueueStart(self::PARTITION, -5, $this->queue);`. Although this shouldn't crash, I think that's not expected usage.
